clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -264,40 +271,133 @@ public boolean rename(Path src, Path dst) throws IOException {
             return false;
         }
 
-        ObjectStats srcStat;
-        ObjectsApi objects = lfsClient.getObjects();
+        LakeFSFileStatus srcStatus;
         try {
-            // Stat src file to get its metadata
-            srcStat = objects.statObject(srcObjectLoc.getRepository(), srcObjectLoc.getRef(),
-                    srcObjectLoc.getPath());
-        } catch (ApiException e) {
-            LOG.error("rename: could not get src object stats. src:{}", src, e);
+            srcStatus = getFileStatus(src);
```

Comment:
Nice -- I like that we do a seemingly excessive "getFileStatus" rather than test for src being a directory... but then pass the result for a file.  So either it's a directory "move" and one more action doesn't matter (and atomicity went out the window), or it's a file move and we needed to get a status anyway.